apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -73,11 +76,30 @@ const trackBlockReplacement = ( originalBlockIds, blocks ) => {
 	} );
 };
 
+/**
+ * Track inner blocks replacement.
+ * This is how Page Templates insert their content into page, by replacing everything that was already there.
+ *
+ * @param {Array} rootClientId id of parent block
+ * @param {object|Array} blocks block instance object or an array of such objects
+ * @returns {void}
+ */
+const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
+	castArray( blocks ).forEach( block => {
+		tracksRecordEvent( 'wpcom_block_inserted', {
+			block_name: block.name,
+			blocks_replaced: false,
+			// isInsertingPageTemplate filter is set by Starter Page Templates
+			from_template_selector: applyFilters( 'isInsertingPageTemplate', false ),
```

Comment:
So by default, this is `false` right? It's odd because when I inserted a Columns block the tracks event was missing `from_template_selector` prop.

<img width="1328" alt="Screenshot 2020-02-13 at 11 35 57" src="https://user-images.githubusercontent.com/444434/74430755-71ffd600-4e55-11ea-8303-7430f2196c2c.png">

But then once the Block inserted it automatically added x2 `core/column` (singular!) Blocks and those did have the `from_template_selector` prop:

<img width="932" alt="Screenshot 2020-02-13 at 11 40 01" src="https://user-images.githubusercontent.com/444434/74430838-978cdf80-4e55-11ea-8445-7227f4088a0f.png">

Reply:
Also...just checking, if a filter isn't registered then I assume anything that hooks into that filter is ignored and doesn't throw an error. Just checking because otherwise if you don't ensure widgets is deployed in advance of this PR landing we're going to have errors.

Reply:
>  if a filter isn't registered then I assume anything that hooks into that filter is ignored

100% correct. it just returns the default value immediately, which is `false`

Reply:
> Columns block the tracks event was missing from_template_selector prop.

When you insert a block manually, it uses a different tracking code path where `from_template_selector` can never be true so that's why the check is not included.

> Block inserted it automatically added x2 core/column (singular!) Blocks and those did have the from_template_selector prop:

It's using the same method template selector uses (`replaceInnerBlocks`) so the tracking will include `from_template_selector` and it should be `false` because no filter is registered at the moment to change it to anything else

Reply:
Thanks for thorough explanations.